src/targets/Logary.Targets.Serilog/Targets_Serilog.fs

```diff
@@ -0,0 +1,169 @@
+    | values ->
+      
+      let objectValues = values
+                         |> List.choose (function | Object v -> Some v | _ -> None)
+      null // TODO:
+
+  let rec logaryValueToSerilogProperty (v : Value) : Serilog.Events.LogEventPropertyValue =
+    let toScalar x = Serilog.Events.ScalarValue x :> Serilog.Events.LogEventPropertyValue
+    match v with
+    | Value.Bool b -> toScalar b      
+    | Value.Float f -> toScalar f
+    | Value.Int64 i -> toScalar i
+    | Value.BigInt i -> toScalar i
+    | Value.String s -> toScalar s
+    | Value.Binary (bytes, _) -> toScalar bytes // TODO: content type? consider this again
+    | Value.Object map ->
+      let stringOrNull s = match s with String tt -> tt | _ -> null
+      let maybeTypeTag = map |> Map.tryPick (fun k v -> if k = "_typeTag" then Some (stringOrNull v) else None)
+      let typeTagOrNull = match maybeTypeTag with Some s -> s | None -> null
+      let serilogProperties =
+        map
+        |> Map.filter (fun k _ -> k <> "_typeTag")
+        |> Map.toSeq
+        |> Seq.map (fun (k, v) -> Serilog.Events.LogEventProperty(k, logaryValueToSerilogProperty v))
+
+      Serilog.Events.StructureValue(serilogProperties, typeTagOrNull)
+      :> Serilog.Events.LogEventPropertyValue
+    | Fraction (x, y) as f -> toScalar f // TODO:
+    | Value.Array items ->
+      Serilog.Events.SequenceValue (items |> List.map logaryValueToSerilogProperty)
+      :> Serilog.Events.LogEventPropertyValue
+
+  let logaryPointNameToSerilog (pointName : PointName) = seq {
+    yield Serilog.Events.LogEventProperty(
+      "SourceLoggerName", Serilog.Events.ScalarValue(pointName.ToString())) }
+
+  let defaultMessageToSerilogProperties (conf : SerilogConf) (logaryMessage : Message) =
+    let newSerilogProperty name value =
+      Serilog.Events.LogEventProperty(name, logaryValueToSerilogProperty value)
+
+    seq {
+      if conf.nameToSerilog.IsSome then
+        yield! conf.nameToSerilog.Value logaryMessage.name
+      
+      if not logaryMessage.fields.IsEmpty then
+        yield!
+          logaryMessage.fields
+          |> Map.toSeq
+          |> Seq.map (fun (pn, Field (v, u)) -> newSerilogProperty (string pn) v)
+
+      if not logaryMessage.context.IsEmpty then
+        yield!
+          logaryMessage.context
+          |> Map.toSeq
+          |> Seq.map (fun (k, v) -> newSerilogProperty k v)
+    }
+
+  let logaryToSerilog (conf : SerilogConf) (logaryMessage : Logary.Message) =
+    let template = match logaryMessage.value with
+                   | Event template -> template
+                   | _ -> failwith "logaryToSerilog only supports PointValue.Event"
+
+    Serilog.Events.LogEvent(
+      timestamp = DateTimeOffset.ofEpoch logaryMessage.timestamp,
+      level = toSerilogLevel logaryMessage.level,
+      ``exception`` = extractFirstExceptionOrNull logaryMessage,
+      messageTemplate = Serilog.Parsing.MessageTemplateParser().Parse template,
```

Comment:
Internally serilog uses a template cache. I haven't yet considered doing the same here.